src/squlearn/encoding_circuit/circuit_library/chebyshev_tower.py

```diff
@@ -57,18 +62,37 @@ def __init__(
         self.rotation_gate = rotation_gate
         self.hadamard_start = hadamard_start
         self.arrangement = arrangement
+        self.nonlinearity = nonlinearity
 
         if self.rotation_gate not in ("rx", "ry", "rz"):
             raise ValueError("Rotation gate must be either 'rx', 'ry' or 'rz'")
 
         if self.arrangement not in ("block", "alternating"):
             raise ValueError("Arrangement must be either 'block' or 'alternating'")
 
+        if self.nonlinearity not in ("arccos", "arctan"):
```

Comment:
We never used it, but in principle, arcsin and tanh are also possible.

Reply:
I could add it, if we want to 🙂